backend/src/modules/doubles-tournament/core/domain/QueuedMatch.ts

```diff
@@ -0,0 +1,70 @@
+import { TeamId } from './TeamId';
+import { TournamentId } from './TournamentId';
+import { DoublesTournament } from './DoublesTournament';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+import { MatchesQueue } from './MatchesQueue';
+import { MatchWasQueued } from './event/MatchWasQueued';
+
+export class QueuedMatch {
+  readonly matchNumber: number;
+  readonly team1Id: TeamId;
+  readonly team2Id: TeamId;
+
+  constructor(props: { matchNumber: number; team1Id: TeamId; team2Id: TeamId }) {
+    this.matchNumber = props.matchNumber;
+    this.team1Id = props.team1Id;
+    this.team2Id = props.team2Id;
+  }
+}
+
+export function pushMatchToQueue(
+  tournament: DoublesTournament | undefined,
+  queue: MatchesQueue | undefined,
+  command: {
+    tournamentId: TournamentId;
+    matchNumber: number;
+    team1Id: TeamId;
+    team2Id: TeamId;
+  },
+  currentTime: Date,
+): DomainCommandResult<MatchesQueue> {
+  if (tournament === undefined) {
+    throw new Error("This tournament doesn't exists.");
+  }
+  if (command.matchNumber <= 0) {
+    throw new Error('Such match number is incorrect!');
+  }
+  if (queue === undefined) {
+    queue = new MatchesQueue({
+      tournamentId: command.tournamentId,
+      queuedMatches: [],
+    });
+  }
+  if (isMatchAlreadyInQueue(command.matchNumber, queue)) {
+    throw new Error('Such match is already waiting in matches queue!');
+  }
+
+  const matchToPush = new QueuedMatch({
+    matchNumber: command.matchNumber,
+    team1Id: command.team1Id,
+    team2Id: command.team2Id,
+  });
+
+  queue.queuedMatches.push(matchToPush);
```

Comment:
Dodatkowo - dobrze jest z tymi funkcjami w domenie dążyć do czystych funkcji, czyli (staryStan, jakiesDaneZCommand) -> nowyStan. Fajnie jakby obiekt MatchesQueue zrobić immutable. Bo teraz staryStan się zmienia jeszcze przed zakończeniem funkcji. 

To jak Wam opowiadałem z liczbami. Jeśli mamy funkcję plus i zrobimy 2 + 3, to wynik jest 5. 
W Twoim przypadku na początku masz 2 + 3 i wynik też jest 5, ale jakby przeczytać to jeszcze raz to:
5 + 3 = 5, bo zmieniasz 2 zamiast zwrócić coś nowego. Szczeglnie już to w ogóle przydatne jest w Javie, gdzie mamy wiele wątków. Wtedy ktoś może Ci w międzyczasie np. podmienić coś w tej liście - w trakcie trwania funkcji i spowodować zły wynik.